hazelcast/src/main/java/com/hazelcast/internal/util/ThreadLocalRandom.java

```diff
@@ -16,199 +16,60 @@
-                    return new ThreadLocalRandom();
-                }
-            };
-
-
-    /**
-     * Constructor called only by localRandom.initialValue.
-     */
-    ThreadLocalRandom() {
-        super();
-        initialized = true;
+    private ThreadLocalRandom() {
     }
 
     /**
-     * Returns the current thread's {@code ThreadLocalRandom}.
+     * Returns the current thread's {@link Random}.
      *
-     * @return the current thread's {@code ThreadLocalRandom}
+     * @return the current thread's {@link Random}.
      */
-    public static ThreadLocalRandom current() {
-        return localRandom.get();
-    }
-
-    /**
-     * Throws {@code UnsupportedOperationException}.  Setting seeds in
-     * this generator is not supported.
-     *
-     * @throws UnsupportedOperationException always
-     */
-    public void setSeed(long seed) {
-        if (initialized)
-            throw new UnsupportedOperationException();
-        rnd = (seed ^ multiplier) & mask;
-    }
-
-    protected int next(int bits) {
-        rnd = (rnd * multiplier + addend) & mask;
-        return (int) (rnd >>> (48-bits));
-    }
-
-    /**
-     * Returns a pseudorandom, uniformly distributed value between the
-     * given least value (inclusive) and bound (exclusive).
-     *
-     * @param least the least value returned
-     * @param bound the upper bound (exclusive)
-     * @return the next value
-     * @throws IllegalArgumentException if least greater than or equal
-     * to bound
-     */
-    public int nextInt(int least, int bound) {
-        if (least >= bound)
-            throw new IllegalArgumentException();
-        return nextInt(bound - least) + least;
-    }
-
-    /**
-     * Returns a pseudorandom, uniformly distributed value
-     * between 0 (inclusive) and the specified value (exclusive).
-     *
-     * @param n the bound on the random number to be returned.  Must be
-     *        positive.
-     * @return the next value
-     * @throws IllegalArgumentException if n is not positive
-     */
-    public long nextLong(long n) {
-        if (n <= 0)
-            throw new IllegalArgumentException("n must be positive");
-        // Divide n by two until small enough for nextInt. On each
-        // iteration (at most 31 of them but usually much less),
-        // randomly choose both whether to include high bit in result
-        // (offset) and whether to continue with the lower vs upper
-        // half (which makes a difference only if odd).
-        long offset = 0;
-        while (n >= Integer.MAX_VALUE) {
-            int bits = next(2);
-            long half = n >>> 1;
-            long nextn = ((bits & 2) == 0) ? half : n - half;
-            if ((bits & 1) == 0)
-                offset += n - nextn;
-            n = nextn;
+    public static Random current() {
+        Random random = THREAD_LOCAL_RANDOM.get();
+        if (random == null) {
+            random = new Random(seedUniquifier() ^ System.nanoTime());
```

Comment:
I just moved this from the `UuidUtil`, so we would have to ask whoever wrote that originally. Since I re-use the new `ThreadLocalRandom` in the `UuidUtil`, I didn't want to change the Random initialization.

I think the explanation was that some JDKs initialize the random with just `System.currentMillis()`, not `System.nanoTime()`. This is still a kind of backport of ThreadLocalRandom, so maybe it's valid to enforce the initialization we prefer.